transport/src/main/java/io/camunda/zeebe/transport/stream/impl/AggregatedClientStream.java

```diff
@@ -126,16 +126,47 @@ private void push(final DirectBuffer buffer) {
       throw new NoSuchStreamException();
     }
 
-    final ClientStream<M> clientStream = pickRandomStream(streams);
-    LOGGER.trace("Pushing data from stream [{}] to client [{}]", streamId, clientStream.streamId());
-    clientStream.clientStreamConsumer().push(buffer);
-  }
-
-  private ClientStream<M> pickRandomStream(final Collection<ClientStream<M>> streams) {
     final var targets = new ArrayList<>(streams);
     final var index = ThreadLocalRandom.current().nextInt(streams.size());
 
-    return targets.get(index);
+    final var result = tryPush(targets, index, 1, buffer);
```

Comment:
:wrench: We could also just generate a random sequence of the indexes and go through these in order.

Reply:
Since the the common case is that the first attempts succeeds, it might be an overhead to generate a random sequence for every request.